app/util/rpc-domain-utils.test.ts

```diff
@@ -1,152 +1,377 @@
+            rpc: ['https://Known-Domain.com/api'],
+          },
+        ];
+        (StorageWrapper.getItem as jest.Mock).mockResolvedValue(JSON.stringify(mockChains));
+        await initializeRpcProviderDomains();
+        // Execute
+        const result1 = isKnownDomain('known-domain.com');
+        const result2 = isKnownDomain('KNOWN-DOMAIN.COM');
+        // Verify
+        expect(result1).toBe(true);
+        expect(result2).toBe(true);
       });
     });
+  });
+  describe('extractRpcDomain', () => {
+    describe('when processing URLs', () => {
+      beforeEach(async () => {
+        resetModuleState();
+        const mockChains: SafeChain[] = [
+          {
+            chainId: '1',
+            name: 'Test Chain',
+            nativeCurrency: { symbol: 'TEST' },
+            rpc: ['https://known-domain.com/api'],
+          },
+        ];
+        (StorageWrapper.getItem as jest.Mock).mockResolvedValue(JSON.stringify(mockChains));
+        await initializeRpcProviderDomains();
+      });
+      it('returns domain for known domains', () => {
+        // Execute
+        const result = extractRpcDomain('https://known-domain.com/api');
+        // Verify
+        expect(result).toBe('known-domain.com');
+      });
+      it('returns Invalid for invalid URLs', () => {
+        // Execute
+        const result = extractRpcDomain(':::invalid-url');
+        // Verify
+        expect(result).toBe('invalid');
+      });
+      it('returns Private for unknown domains', () => {
+        // Execute
+        const result = extractRpcDomain('https://unknown-domain.com');
+        // Verify
+        expect(result).toBe('private');
+      });
+      it('returns actual domain for Infura URLs', () => {
+        // Execute
+        const result = extractRpcDomain('https://mainnet.infura.io');
+        // Verify
+        expect(result).toBe('mainnet.infura.io');
+      });
+      it('returns actual domain for Alchemy URLs', () => {
+        // Execute
+        const result = extractRpcDomain('https://eth-mainnet.alchemyapi.io');
+        // Verify
+        expect(result).toBe('eth-mainnet.alchemyapi.io');
+      });
+      it('returns Private for localhost', () => {
+        // Execute
+        const result1 = extractRpcDomain('http://localhost:8545');
+        const result2 = extractRpcDomain('http://127.0.0.1:8545');
 
-    it('returns { properties: { rpc_domain: "invalid" }, sensitiveProperties: {} } for an invalid domain', () => {
-      jest.spyOn(rpcUtils, 'getNetworkRpcUrl').mockReturnValue('invalid-url');
-      jest.spyOn(rpcUtils, 'extractRpcDomain').mockReturnValue(RpcDomainStatus.Invalid);
-      expect(generateRPCProperties('0x2')).toEqual({
-        properties: { rpc_domain: RpcDomainStatus.Invalid },
-        sensitiveProperties: {},
+        // Verify
+        expect(result1).toBe('private');
+        expect(result2).toBe('private');
+      });
+      it('handles URLs without protocol', () => {
```

Comment:
S: avoid "handle" and explain what's actually being done here:

```suggestion
      it('returns the domain for URLs without protocol', () => {
```